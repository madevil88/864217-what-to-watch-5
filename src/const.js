const InitialState = {
  GENRE: `All genres`,
  FILMS_COUNT: 8
};

const Genre = {
  ANIMATED_CARTOON: `Animated cartoon`,
  BIOGRAPHICAL: `Biographical`,
  COMEDY: `Comedy`,
  DRAMA: `Drama`,
  FANTASY: `Fantasy`,
  THRILLER: `Thriller`,
};

const FilmStatus = {
  AWESOME: `Awesome`,
  VERY_GOOD: `Very good`,
  GOOD: `Good`,
  NORMAL: `Normal`,
  BAD: `Bad`,
};

const SHOW_GENRES_COUNT = 10;

const TabsMap = {
  0: `Overview`,
  1: `Details`,
  2: `Reviews`
};

const BACKEND_URL = `https://5.react.pages.academy/wtw`;
const REQUEST_TIMEOUT = 5000;

const HttpCode = {
  UNAUTHORIZED: 401,
  NOT_FAUND: 404,
  BAD_REQUEST: 400
};

const NameSpace = {
  DATA: `DATA`,
  APP_STATE: `APP_STATE`,
  USER: `USER`,
};

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const AppRoute = {
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  FILM: `/films/:id`,
  ROOT: `/`,
  ADD_REVIEW: `/films/:id/review`,
  PLAYER: `/player/:id`,
};

const APIRoute = {
  FILMS: `/films`,
  LOGIN: `/login`,
  PROMO_FILM: `/films/promo`,
  FAVORITE_FILMS: `/favorite`,
};

const ReviewTextLength = {
  MIN: 50,
  MAX: 400
};

const ErrorMessageMap = {
  400: `You made a mistake, correct and send again`,
  401: `You are not logged in, please log in`
};

const RexExp = {
  EMAIL: /^[a-zA-Z0-9.!#$%&'*+/=?^_\`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
};

export {InitialState,
  Genre,
  FilmStatus,
  TabsMap,
  SHOW_GENRES_COUNT,
  BACKEND_URL,
  REQUEST_TIMEOUT,
  HttpCode,
  NameSpace,
  AuthorizationStatus,
  AppRoute,
  APIRoute,
  ReviewTextLength,
  ErrorMessageMap,
  RexExp};
