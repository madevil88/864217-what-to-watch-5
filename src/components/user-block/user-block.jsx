import React from "react";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import {AppRoute, AuthorizationStatus} from "../../const";
import {getAuthorizationStatus} from "../../store/selectors";

const UserBlock = (props) => {
  const {authorizationStatus} = props;
  if (authorizationStatus === AuthorizationStatus.AUTH) {
    return (
      <div className="user-block">
        <div className="user-block__avatar">
          <img src="../../img/avatar.jpg" alt="User avatar" width="63" height="63" />
        </div>
      </div>
    );
  } else {
    return (
      <div className="user-block">
        <Link to={AppRoute.LOGIN}>
          <span className="user-block__link">Sign in</span>
        </Link>
      </div>
    );
  }
};

UserBlock.propTypes = {};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
});

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);

