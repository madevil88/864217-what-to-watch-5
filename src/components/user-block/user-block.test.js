import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {UserBlock} from "./user-block";
import {AuthorizationStatus} from "../../const";

const renderer = new ShallowRenderer();

it(`Should UserBlock render correctly`, () => {
  renderer.render(
      <UserBlock
        authorizationStatus={AuthorizationStatus.AUTH}
      />
  );

  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
