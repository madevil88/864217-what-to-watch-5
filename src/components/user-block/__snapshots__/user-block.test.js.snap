// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserBlock render correctly 1`] = `
<div
  className="user-block"
>
  <Link
    to="/mylist"
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="../../img/avatar.jpg"
        width="63"
      />
    </div>
  </Link>
</div>
`;
